policies/rudderc/src/backends/unix/ncf/method_call.rs

```diff
@@ -173,9 +173,25 @@ pub fn method_call(
         "args".to_string(),
         "class_prefix".to_string(),
     ];
-    if let Condition::Expression(_) = condition {
-        call_parameters.push(cfengine_canonify_condition(condition.as_ref()));
-        method_parameters.push("method_call_condition".to_string())
+    // If the item is a result of a foreach loop, we must assume that one of the branch could define
+    // a condition, and so, each branch should call the bundle using the method_call_condition
+    match condition {
+        Condition::Expression(_) => {
+            call_parameters.push(cfengine_canonify_condition(condition.as_ref()));
```

Comment:
You should be able to to just remove the if and keep the same content. `Condition`'s `Display` implementation already handle the true/false cases.